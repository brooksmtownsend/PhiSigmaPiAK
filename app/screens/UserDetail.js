import React, { Component } from 'react';
import { ScrollView } from 'react-native';
import { Tile, List, ListItem } from 'react-native-elements';

class UserDetail extends Component {
  render() {
    const params = this.props.navigation.state.params;
    const name = params.FirstName + " " + params.LastName;
    const pronouns = params.PreferredPronouns
    const email = params.Email;
    const phone = params.PhoneNumber;
    const iClass = params.InitiateClass;
    const picture = params.Picture;

    return (
      <ScrollView>
        {<Tile
          imageSrc={{ uri: picture}}
          featured
          title={name}
          caption={email}
        />}

        <List>
          <ListItem
            title="Initiate Class"
            rightTitle={iClass}
            hideChevron
          />
          <ListItem
            title="Preferred Pronouns"
            rightTitle={pronouns}
            hideChevron
          />
        </List>

        <List>
          <ListItem
            title="Email"
            rightTitle={email}
            hideChevron
          />
          <ListItem
            title="Phone"
            rightTitle={phone}
            hideChevron
          />
        </List>

        {/* <List>
          <ListItem
            title="Username"
            rightTitle={login.username}
            hideChevron
          />
        </List>

        <List>
          <ListItem
            title="Birthday"
            rightTitle={dob}
            hideChevron
          />
          <ListItem
            title="City"
            rightTitle={location.city}
            hideChevron
          />
        </List> */}
      </ScrollView>
    );
  }
}

export default UserDetail;
