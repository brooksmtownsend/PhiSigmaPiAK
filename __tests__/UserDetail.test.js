var React = require('react');
var ReactNative = require('react-native');
var renderer = require('react-test-renderer');
var jest = require('jest');

import UserDetail from '../app/screens/UserDetail'

describe("UserDetail Functionality", () => {
  it("renders correctly", () => {
    const userDetail = renderer.create(
      <UserDetail navigation={{state: {params:{
        FirstName:"first",
        LastName:"last",
        PreferredPronouns:"h/h/h",
        Email:"a@a.a",
        PhoneNumber:"9999999999",
        InitiateClass:"Alpha",
        PictureURL:"https://i.imgur.com/RwrMEtI.jpg"
      }}}} />).toJSON()
    expect(userDetail).toMatchSnapshot()
  })
})
