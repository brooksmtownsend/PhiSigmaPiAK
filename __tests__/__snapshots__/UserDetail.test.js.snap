// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail Functionality renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 600,
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "height": 600,
              "justifyContent": "center",
              "width": 750,
            },
            undefined,
          ]
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "uri": "https://i.imgur.com/RwrMEtI.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "rgba(0,0,0,0.2)",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "paddingBottom": 40,
                "paddingLeft": 25,
                "paddingRight": 25,
                "paddingTop": 45,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                Object {
                  "fontSize": 27.5,
                },
                undefined,
                undefined,
                undefined,
                Object {},
                undefined,
                Array [
                  Object {
                    "backgroundColor": "rgba(0,0,0,0)",
                    "color": "#ffffff",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "backgroundColor": "rgba(0,0,0,0)",
                    "color": "#ffffff",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#bbb",
            "borderTopWidth": 1,
            "marginTop": 20,
          },
          undefined,
        ]
      }
    >
      <View
        disabled={false}
        onLongPress={undefined}
        onPress={undefined}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#bbb",
              "borderBottomWidth": 1,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            undefined,
            false,
            false,
          ]
        }
        underlayColor="white"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Initiate Class
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#bdc6cf",
                      "marginRight": 5,
                    },
                    Object {
                      "color": "#444",
                    },
                  ],
                ]
              }
            >
              Alpha
            </Text>
          </View>
        </View>
      </View>
      <View
        disabled={false}
        onLongPress={undefined}
        onPress={undefined}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#bbb",
              "borderBottomWidth": 1,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            undefined,
            false,
            false,
          ]
        }
        underlayColor="white"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Preferred Pronouns
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#bdc6cf",
                      "marginRight": 5,
                    },
                    Object {
                      "color": "#444",
                    },
                  ],
                ]
              }
            >
              h/h/h
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#bbb",
            "borderTopWidth": 1,
            "marginTop": 20,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Email
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#bdc6cf",
                      "marginRight": 5,
                    },
                    Object {
                      "color": "#444",
                    },
                  ],
                ]
              }
            >
              a@a.a
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Phone
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#bdc6cf",
                      "marginRight": 5,
                    },
                    Object {
                      "color": "#444",
                    },
                  ],
                ]
              }
            >
              9999999999
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
